### PR TITLE
[LOOP-4484] Inject future values with TestingCGMManager

### PR DESCRIPTION
**Pull request by Camji55**
https://tidepool.atlassian.net/browse/LOOP-4484

### CODE REVIEW

File: LoopTestingKit/TestingScenario.swift [hunk 1]
```diff
@@ -18,7 +18,8 @@ public struct TestingScenario {
     public func instantiate(relativeTo referenceDate: Date = Date()) -> TestingScenarioInstance {
         let glucoseSamples = dateRelativeGlucoseSamples
             .map { $0.newGlucoseSample(relativeTo: referenceDate) }
-            .filter { $0.date <= referenceDate }
+        let pastGlucoseSamples = glucoseSamples.filter { $0.date <= referenceDate }
+        let futureGlucoseSamples = glucoseSamples.filter { $0.date > referenceDate }
```
Comment (nhamming) [inline LoopTestingKit/TestingScenario.swift:22]: It looks like the glucoseSamples already filter based on the referenceDate? Should `dateRelativeGlucoseSamples` be used?

File: MockKit/MockCGMManager.swift [hunk 2]
```diff
@@ -568,9 +568,9 @@ public final class MockCGMManager: TestingCGMManager {
         }
     }
 
-    public func injectGlucoseSamples(_ samples: [NewGlucoseSample]) {
-        guard !samples.isEmpty else { return }
-        sendCGMReadingResult(CGMReadingResult.newData(samples.map { NewGlucoseSample($0, device: device) } ))
+    public func injectGlucoseSamples(_ pastSamples: [NewGlucoseSample], futureSamples: [NewGlucoseSample]) {
+        guard !pastSamples.isEmpty else { return }
+        sendCGMReadingResult(CGMReadingResult.newData(pastSamples.map { NewGlucoseSample($0, device: device) } ))
```
Comment (nhamming) [inline MockKit/MockCGMManager.swift:573]: Should the future samples be used here?

**Review by nhamming — COMMENTED** [review]
Couple of questions.

File: LoopTestingKit/TestingScenario.swift (hunk 1 above)
Comment (Camji55) [reply LoopTestingKit/TestingScenario.swift:22]: The `DateRelativeGlucoseSample` is just a structured type used in the scenario file and needs to be converted to a `NewGlucoseSample` with the dateOffset converted to a relative date which is what the line above this does.

File: MockKit/MockCGMManager.swift (hunk 2 above)
Comment (Camji55) [reply MockKit/MockCGMManager.swift:573]: Nope, I left the existing behavior alone for the CGM Simulator. I made a note in the ticket regarding creating a new ticket to bring this new functionality to the CGM Simulator.

https://tidepool.atlassian.net/browse/LOOP-4484

File: LoopTestingKit/TestingScenario.swift (hunk 1 above)
Comment (nhamming) [reply LoopTestingKit/TestingScenario.swift:22]: Oh... just realized I mis-read. all good.

**Review by nhamming — APPROVED** [review]
LGTM

**Pull request merged**